python/shotgun_model/shotgun_hierarchy_item.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+
+from .shotgun_standard_item import ShotgunStandardItem
+
+
+class ShotgunHierarchyItem(ShotgunStandardItem):
+    """
+    A subclass of ``ShotgunStandardItem`` with access to data provided via
```

Comment:
what's the deal construction? Can this be constructed by any process or only by factory methods? I think we should control creation so we can control memory allocation and release. Items inserted via 

https://github.com/shotgunsoftware/tk-framework-shotgunutils/pull/49/files#diff-d442a3297316abbae14a00adf3457696R548

Should probably be created via factory of some sort.